airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftSqlOperations.java

```diff
@@ -64,4 +99,81 @@ public boolean isValidData(final JsonNode data) {
+        .toList();
+    if (!schemaAndTableWithNotSuperType.isEmpty()) {
+      updateVarcharDataColumnToSuperDataColumn(database, schemaAndTableWithNotSuperType);
+    }
+  }
+
+  /**
+   * @param database   - Database object for interacting with a JDBC connection.
+   * @param schemaName - schema to update.
+   */
+  private List<String> discoverNotSuperTables(final JdbcDatabase database,
+      final String schemaName) {
+    List<String> schemaAndTableWithNotSuperType = new ArrayList<>();
+    try {
+      LOGGER.info("Selecting table types...");
+      database.execute(String.format("set search_path to %s", schemaName));
+      final List<JsonNode> tablesNameWithoutSuperDatatype = database.bufferedResultSetQuery(
+          conn -> conn.createStatement().executeQuery(String.format(SELECT_ALL_TABLES_WITH_NOT_SUPER_TYPE_SQL_STATEMENT,
+              schemaName,
+              JavaBaseConstants.COLUMN_NAME_DATA,
+              JavaBaseConstants.COLUMN_NAME_EMITTED_AT,
+              JavaBaseConstants.COLUMN_NAME_AB_ID)),
+          getDefaultSourceOperations()::rowToJson);
+      if (tablesNameWithoutSuperDatatype.isEmpty()) {
+        return Collections.emptyList();
+      } else {
+        tablesNameWithoutSuperDatatype.forEach(e ->
+            schemaAndTableWithNotSuperType.add(e.get("schemaname").textValue() + "." + e.get("tablename").textValue()));
+        return schemaAndTableWithNotSuperType;
+      }
+    } catch (SQLException e) {
+      LOGGER.error("Error during discoverNotSuperTables() appears: ", e);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return Collections.emptyList();
+  }
+
+  /**
+   * We prepare one query for all tables with not super type for updating.
+   *
+   * @param database                       - Database object for interacting with a JDBC connection.
+   * @param schemaAndTableWithNotSuperType - list of tables with not super type.
+   */
+  private void updateVarcharDataColumnToSuperDataColumn(final JdbcDatabase database, final List<String> schemaAndTableWithNotSuperType) {
+    LOGGER.info("Updating VARCHAR data column to super...");
+    StringBuilder finalSqlStatement = new StringBuilder();
+    // To keep the previous data, we need to add next columns: _airbyte_data, _airbyte_emitted_at
+    // We do such workflow because we can't directly CAST VARCHAR to SUPER column. _airbyte_emitted_at column recreated to keep
+    // the COLUMN order. This order is required to INSERT the values in correct way.
+    schemaAndTableWithNotSuperType.forEach(schemaAndTable -> {
+      LOGGER.info("Altering table {} column _airbyte_data to SUPER.", schemaAndTable);
+      finalSqlStatement.append(String.format(ALTER_TMP_TABLES_WITH_NOT_SUPER_TYPE_TO_SUPER_TYPE,
+          schemaAndTable,
+          JavaBaseConstants.COLUMN_NAME_DATA,
+          JavaBaseConstants.COLUMN_NAME_EMITTED_AT));
+    });
+    try {
+      database.execute(finalSqlStatement.toString());
```

Comment:
should this be executed in a transaction?

Reply:
@girarda Does it really make sense to execute this statement in `database.executeWithinTransaction()` ?
As far as I know: airbyte interacts with DB connection in `auto-commit mode = true`. And when the connection will be closed these changes will be committed. One thing to point out is this StringBuilder contains SQL statements for all tables required migration. Thanks.